lib/p2p/Peer.ts

```diff
@@ -748,35 +748,32 @@ class Peer extends EventEmitter {
       assert(this.expectedNodePubKey);
       await this.initSession(ownNodeState, nodeKey, this.expectedNodePubKey!);
       const sessionInit = await this.waitSessionInit();
-      this.ackSession(sessionInit, nodeKey);
+      return this.ackSession(sessionInit, nodeKey);
     } else {
       // inbound handshake
       const sessionInit = await this.waitSessionInit();
-      this.ackSession(sessionInit, nodeKey);
-      await this.initSession(ownNodeState, nodeKey, sessionInit.body!.nodeState.nodePubKey);
+      await this.ackSession(sessionInit, nodeKey);
+      return this.initSession(ownNodeState, nodeKey, sessionInit.body!.nodeState.nodePubKey);
```

Comment:
It might be simpler to do `await` instead of `return`, since that's how this function was structured. Unless you'll do `return` on the first `if`, and remove the `else`. But in this case, I think if/else makes more sense. 

Also, in many places in this PR you're doing `return` instead of `await`. But if the returned value is void, it might be less confusing to simply do `await`.

Reply:
I definitely agree in this example, and in regards to other places where I'm returning the `Promise<void>` explicitly rather than `await`, I did it mainly because it required fewer lines changed because I didn't have to make the functions `async`. But you may be right that the code is clearer without any `return` statements in those methods that aren't intended to return anything. Functionally they are the same, but if you still think it's a good idea to use `await` instead of `return` let me know and I'll change them elsewhere.

Reply:
If it's just about adding `async` to the function, i'd say it's preferred over using `return`, which I found confusing.